src/main/java/com/epam/ld/module2/testing/Messenger.java

```diff
@@ -28,10 +28,21 @@ public Messenger(MailServer mailServer,
      *
      * @param client   the client
      * @param template the template
+     * @return String
      */
-    public void sendMessage(Client client, Template template) {
-        String messageContent =
-            templateEngine.generateMessage(template, client);
-        mailServer.send(client.getAddresses(), messageContent);
+    public String sendMessage(Client client, Template template) {
+        if(client != null && template != null) {
+            String messageContent =
+                    templateEngine.generateMessage(template, client);
+            mailServer.send(client.getAddresses(), messageContent);
+
+            return messageContent;
+        }
+
+        else if(client == null) {
+            return null;

```

Comment:
Do we really need to return something here? It isn't used anywhere, except tests. 
Besides returning null from methods is usually a bad idea

Reply:
Refactored this